sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c

```diff
@@ -0,0 +1,447 @@
+  uint32_t data = *((uint32_t *)kErrorRamAddress);
+
+  // This normally should not run.
+  LOG_INFO("Read from address 0x%0x with expected error gets 0x%x",
+           kErrorRamAddress, data);
+
+  wait_for_interrupt();
+  CHECK(false, "This should not be reached");
+}
+
+bool test_main(void) {
+  // Enable global and external IRQ at Ibex.
+  irq_global_ctrl(true);
+  irq_external_ctrl(true);
+
+  init_peripherals();
+
+  // Enable all the interrupts used in this test.
+  rv_plic_testutils_irq_range_enable(
+      &plic, kPlicTarget, kTopEarlgreyPlicIrqIdAonTimerAonWkupTimerExpired,
+      kTopEarlgreyPlicIrqIdAonTimerAonWdogTimerBark);
+  rv_plic_testutils_irq_range_enable(&plic, kPlicTarget,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassa,
+                                     kTopEarlgreyPlicIrqIdAlertHandlerClassd);
+
+  // Enable access to flash for storing info across resets.
+  LOG_INFO("Setting default region accesses");
+  flash_ctrl_testutils_default_region_access(&flash_ctrl_state,
+                                             /*rd_en*/ true,
+                                             /*prog_en*/ true,
+                                             /*erase_en*/ true,
+                                             /*scramble_en*/ false,
+                                             /*ecc_en*/ false,
+                                             /*he_en*/ false);
+
+  // Get the flash maintained reset counter.
+  reset_count = flash_ctrl_testutils_counter_get(kCounterReset);
+  LOG_INFO("Reset counter value: %u", reset_count);
+  if (reset_count > kMaxResets) {
+    CHECK(false, "Got too many resets (%d)", reset_count);
+  }
+
+  // Increment reset counter to know where we are.
+  flash_ctrl_testutils_counter_set_at_least(&flash_ctrl_state, kCounterReset,
+                                            reset_count + 1);
+
+  // Check if there was a HW reset caused by the escalation.
+  dif_rstmgr_reset_info_bitfield_t rst_info;
+  rst_info = rstmgr_testutils_reason_get();
+  rstmgr_testutils_reason_clear();
+
+  CHECK(rst_info == kDifRstmgrResetInfoPor ||
+            rst_info == kDifRstmgrResetInfoEscalation,
+        "Wrong reset reason %02X", rst_info);
+
+  if (rst_info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Booting for the first time, starting test");
+    execute_test();
+  } else if (rst_info == kDifRstmgrResetInfoEscalation) {
+    LOG_INFO("Booting for the second time due to escalation reset");
+
+    int interrupt_count = flash_ctrl_testutils_counter_get(kCounterInterrupt);
+    // TODO: settle the interrupt could
+    LOG_INFO("Interrupt count %d", interrupt_count);
+    //    CHECK(interrupt_count > 0, "Expected at least one regular interrupt");
+
+    int nmi_interrupt_count = flash_ctrl_testutils_counter_get(kCounterNmi);
+    // TODO: settle the nmi could
+    LOG_INFO("NMI count %d", interrupt_count);
+    //    CHECK(nmi_interrupt_count > 0, "Expected at least one nmi");
```

Comment:
these 2 checks seem valid. can we uncomment them?

Reply:
Only the nmi is really required, since the regular interrupt can be "defeated" by the NMI since the threshold is at 0.